test/extensions/revealable.ts

```diff
@@ -0,0 +1,98 @@
+import { ethers } from 'hardhat';
+import { expect } from 'chai';
+import { loadFixture } from '@nomicfoundation/hardhat-network-helpers';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import { IOtherInterface, IRMRKRevealable, IRMRKRevealer, IERC5773 } from '../interfaces';
+import { RMRKMultiAssetRevealableMock, RMRKRevealerMock } from '../../typechain-types';
+import { bn } from '../utils';
+
+const UNREVEALED_ASSET_ID = 1;
+const REVEALED_ASSET_ID = 2;
+
+async function revealableFixture(): Promise<{
+  revealable: RMRKMultiAssetRevealableMock;
+  revealer: RMRKRevealerMock;
+  deployer: SignerWithAddress;
+  user1: SignerWithAddress;
+  user2: SignerWithAddress;
+}> {
+  const [deployer, user1, user2] = await ethers.getSigners();
+  const revealableFactory = await ethers.getContractFactory('RMRKMultiAssetRevealableMock');
+  const revealerFactory = await ethers.getContractFactory('RMRKRevealerMock');
+  const revealable = await revealableFactory.deploy();
+  await revealable.deployed();
+
+  const revealer = await revealerFactory.deploy(REVEALED_ASSET_ID);
+  await revealer.deployed();
+
+  await revealable.setRevealer(revealer.address);
+
+  await revealable.mint(user1.address, 1);
+  await revealable.mint(user1.address, 2);
+  await revealable.mint(user2.address, 3);
+
+  await revealable.addAssetEntry(UNREVEALED_ASSET_ID, 'ipfs://unrevealed');
+  await revealable.addAssetEntry(REVEALED_ASSET_ID, 'ipfs://revealed');
+
+  await revealable.addAssetToToken(1, UNREVEALED_ASSET_ID, 0);
+  await revealable.addAssetToToken(2, UNREVEALED_ASSET_ID, 0);
+  await revealable.addAssetToToken(3, REVEALED_ASSET_ID, 0);
+
+  await revealable.connect(user1).acceptAsset(1, 0, UNREVEALED_ASSET_ID);
+  await revealable.connect(user1).acceptAsset(2, 0, UNREVEALED_ASSET_ID);
+  await revealable.connect(user2).acceptAsset(3, 0, REVEALED_ASSET_ID);
+
+  return { revealable, revealer, deployer, user1, user2 };
+}
+
+// --------------- TESTS -----------------------
+
+describe('RMRKTokenPropertiesMock', async function () {
```

Comment:
Would be nice to add test with replace id, also with contract where there's more than 1 revealable asset

Reply:
Actually, the tests already uses replace id. Also, I'm focusing on revealable logic which is the abstract contract I'm implementing. 
Revealer here is just a mock to prove the concept, in the future we may add a few implementations and only then I would add more specific cases.

Reply:
Sure, but tests and mocks are like a documentation in this case, so good to have it to cover many cases and be descriptive